PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/ManifestFileUtilsEx.java

```diff
@@ -0,0 +1,196 @@
+            @Override
+            public void windowOpened(WindowEvent e) {
+                // remove old jar and add new jar to project dependency
+                WriteAction.runAndWait(() -> addJarToModuleDependency(jarFile));
+                super.windowOpened(e);
+            }
+            @Override
+            public void windowClosed(WindowEvent e) {
+                WriteAction.runAndWait(() -> removeModuleAndJar());
+                super.windowClosed(e);
+            }
+        });
+        chooser.showDialog();
+        return chooser.getSelected();
+    }
+
+    private static class JarClassFilter implements ClassFilter {
+        final private String filePath;
+        public JarClassFilter(@NotNull String filePath) {
+            this.filePath = filePath;
+        }
+        public boolean isAccepted(final PsiClass aClass) {
+            return ReadAction.compute(() ->
+                    aClass.getContainingFile().getVirtualFile().getPath().startsWith(filePath));
+        }
+    }
+
+    @NotNull
+    private String suggestModuleName(@NotNull String baseName) {
+        String candidate = baseName;
+        int idx = 1;
+        while (ModuleManager.getInstance(myProject).findModuleByName(candidate) != null) {
+            candidate = baseName + (idx++);
+        }
+        return candidate;
+    }
+
+    @NotNull
+    private Module createModule() {
+        String moduleName = suggestModuleName(LOCAL_ARTIFACT_MODULE_NAME);
+        File moduleFile = new File(myProject.getBasePath(), String.format("%s/%s.iml", moduleName, moduleName));
+        Module module = ModuleManager.getInstance(myProject).newModule(moduleFile.getPath(), ModuleTypeId.JAVA_MODULE);
+        return module;
+    }
+
+    private void addJarToModuleDependency(@NotNull VirtualFile jarFile) {
+        try {
+            final Module module = createModule();
+            final List<OrderRoot> myRoots = RootDetectionUtil.detectRoots(
+                    Arrays.asList(jarFile), null, myProject, new DefaultLibraryRootsComponentDescriptor());
+            final ModifiableRootModel modifiableModel = ModuleRootManager.getInstance(module).getModifiableModel();
+            // find a unique library name in the module
+            final String libraryName = LibraryEditingUtil.suggestNewLibraryName(
+                    modifiableModel.getModuleLibraryTable().getModifiableModel(), jarFile.getName());
+            // add library to the model of the new-created module
+            LibrariesContainerFactory.createContainer(modifiableModel).createLibrary(
+                    libraryName, LibrariesContainer.LibraryLevel.MODULE, myRoots);
+            modifiableModel.commit();
+
+            setLocalArtifactModuleName(module.getName());
+            setLocalArtifactLibraryName(libraryName);
+        } catch (Exception ex) {
+            log().warn(String.format("Error add user selected jar(%s) to module dependency. Error:%s", jarFile.getPath(), ex.toString()));
```

Comment:
Wording, the message could be `Failed to add the user selected jar(%s) into module dependency: %s` since the level is `Warning`, we'd better to avoid `Error` in the message.